tools/peptideshaker/macros.xml

```diff
@@ -1,179 +0,0 @@
-<macros>
-    <xml name="stdio">
-        <stdio>
-            <exit_code range="1:" level="fatal" description="Job Failed" />
-            <regex match="java.*Exception" level="fatal" description="Java Exception"/>
-            <regex match="Could not create the Java virtual machine" level="fatal" description="JVM Error"/>
```

Comment:
I just stumbled over the following error in the current version of the SearchGUI tool: 

```
Error occurred during initialization of VM
Could not reserve enough space for 6291456KB object heap
```

Can you add another regex match for recognizing the out of memory condition? A question for @bgruening are the regex conditions processed in order .. and is the exit_code condition checked before the regex conditions? Then you would need to add it in front of the `Could not create the Java virtual machine` regex. 

If you would not like to include this in this PR I'm happy to open a separate PR.

Reply:
FYI: just use level="fatal_oom"

Reply:
Thank you for the suggestion, I will add it to macros_basic.xml as soon as possible (macros.xml will disappear after this PR).
By the way, as you suggested, the exit_code condition is checked before regex:

"" (from https://docs.galaxyproject.org/en/latest/dev/schema.html):
A note should be made on the order in which exit codes and regular expressions are applied and how the processing stops. Exit code rules are applied before regular expression rules. The rationale is that exit codes are more clearly defined and are easier to check computationally, so they are applied first. Exit code rules are applied in the order in which they appear in the tool’s configuration file, and regular expressions are also applied in the order in which they appear in the tool’s configuration file. However, once a rule is triggered that causes a fatal error, no further rules are checked.
""

Reply:
In this case (exit_code is checked before regex) the regex tests seem to make no sense for this tool since searchgui seems to return exit code 1 anyway (at least for the OOM error). 

@bgruening any comment on that? I guess in the light of the recently introduced OOM checks, testing for an exit_code first seems to be not the best idea...?